### PR TITLE
Fix linting error

### LINKED ISSUE #352: Problem with saving into NeXus format including metadata
Somehow I have a problem to use prc.save("*.nxs").
In particular, it complains with e.g.
"The data entry corresponding to /ENTRY[entry]/INSTRUMENT[instrument]/energy_resolution is required and hasn't been supplied by the reader."
Although this information is provided in the JSON file,
but it became a nested JSON file after last changes with #336 changes.

If I am changing file structure from nested
json_data["/ENTRY[entry]/INSTRUMENT[instrument]"]["energy_resolution"]
to simple JSON structure
json_data["/ENTRY[entry]/INSTRUMENT[instrument/energy_resolution"]
then everything is fine.

@rettigl maybe you can help me with this?

### PR DESCRIPTION
**Pull request by rettigl**
downgrade ruff, and update ruff settings
update pynxtools to pynxtools-mpes

Closes #352

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
     branches:
       - main
     tags:
-      - '*'
+      - v[0-9]+.[0-9]+.[0-9]+
```
Comment (zain-sohail) [inline .github/workflows/release.yml:14]: Does this restrict the versioning to between 0 and 9?
If so, why is it placed? Semantic versioning allows for values above 9 as well. Good example is python 3.11.

**Review by zain-sohail — APPROVED** [review]
Only the suggested comment. Otherwise, this can be merged.

File: .github/workflows/release.yml (hunk 1 above)
Comment (rettigl) [reply .github/workflows/release.yml:14]: [0-9]+ is any number of digits, so this should be fine, no? I tested it, however, only for the single-digit case

Comment (zain-sohail) [reply .github/workflows/release.yml:14]: I am not knowing much about regex but I looked it up and it seems good

**Pull request merged**